include/contractor/contractor.hpp

```diff
@@ -833,6 +852,50 @@ class Contractor
                 continue;
             }
 
+            // Check for Potential Self-Loops, only add if self-loop is the shortest loop
+            for (auto out_edge : contractor_graph->GetAdjacentEdgeRange(node))
+            {
+                const ContractorEdgeData &out_data = contractor_graph->GetEdgeData(out_edge);
+                if (out_data.forward)
+                {
+                    const NodeID target = contractor_graph->GetTarget(out_edge);
+                    if (source == target)
+                    {
+                        // TODO we could prune some of these self-loops if we new about the weight
+                        // along the
+                        // edge represented by `node` itself. If the self-loop is longer than the
+                        // edge,
+                        // the self loop can be pruned as well
+                        // Check for Self Loops
+                        const int this_loop_distance = in_data.distance + out_data.distance;
+                        int self_loop_distance = FindSelfLoop(source, this_loop_distance, data);
+                        if (self_loop_distance == this_loop_distance)
+                        {
+                            if (RUNSIMULATION)
+                            {
+                                BOOST_ASSERT(stats != nullptr);
+                                stats->edges_added_count += 2;
+                                stats->original_edges_added_count +=
+                                    2 * (out_data.originalEdges + in_data.originalEdges);
+                            }
+                            else
+                            {
+                                inserted_edges.emplace_back(source, target, this_loop_distance,
+                                                            out_data.originalEdges +
+                                                                in_data.originalEdges,
+                                                            node, true, true, false);
+
+                                inserted_edges.emplace_back(target, source, this_loop_distance,
+                                                            out_data.originalEdges +
+                                                                in_data.originalEdges,
+                                                            node, true, false, true);
```

Comment:
This and above is hard to follow as I have to 1/ look for inserted_edges' value type, 2/ then check which position in the ContractorEdgeData constructor these values correspond to. Hmm probably better to symbolize them, that is create some named booleans and then pass those (especially because only source/target and backward changes).

Would be great to have named parameters á la: fn(forward: true, backward: false) --- oh well :/